gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/AbstractGemFireDistributionActivation.java

```diff
@@ -640,6 +644,7 @@ protected final Object operateOnNonPartitionedRegion(
           (isForSelectForUpdateQuery && this.qInfo.needKeysForSelectForUpdate()),
           this.pvs, rgn, null, this.qInfo.isInsertAsSubSelect(), lcc, this
               .getTimeOutMillis());
+      functionMsg.setAbortOnLowMemory(GfxdHeapThresholdListener.isCancellableQuery(this));
       setAllAreNonPartitionedRegion(rs, isVTIInvolved);
```

Comment:
Looking at different GfxdFunctionMessages, I think StatementCloseExecutorMessage can also be excluded from cancellation.